integration-cli/docker_api_swarm_config_test.go

```diff
@@ -116,3 +116,100 @@ func (s *DockerSwarmSuite) TestAPISwarmConfigsUpdate(c *check.C) {
 	c.Assert(err, checker.IsNil, check.Commentf(string(out)))
 	c.Assert(status, checker.Equals, http.StatusBadRequest, check.Commentf("output: %q", string(out)))
 }
+
+func (s *DockerSwarmSuite) TestAPISwarmTemplatedConfig(c *check.C) {
+	d := s.AddDaemon(c, true, true)
+
+	referencedSecretSpec := swarm.SecretSpec{
+		Annotations: swarm.Annotations{
+			Name: "referencedsecret",
+		},
+		Data: []byte("this is a secret"),
+	}
+
+	referencedSecretID := d.CreateSecret(c, referencedSecretSpec)
+	c.Assert(referencedSecretID, checker.Not(checker.Equals), "", check.Commentf("secrets: %s", referencedSecretID))
+
+	referencedConfigSpec := swarm.ConfigSpec{
+		Annotations: swarm.Annotations{
+			Name: "referencedconfig",
+		},
+		Data: []byte("this is a config"),
+	}
+
+	referencedConfigID := d.CreateConfig(c, referencedConfigSpec)
+	c.Assert(referencedConfigID, checker.Not(checker.Equals), "", check.Commentf("configs: %s", referencedConfigID))
+
+	configSpec := swarm.ConfigSpec{
+		Annotations: swarm.Annotations{
+			Name: "templated_config",
+		},
+		Templating: &swarm.Driver{
+			Name: "golang",
+		},
+		Data: []byte("SERVICE_NAME={{.Service.Name}}\n" +
+			"{{secret \"referencedsecrettarget\"}}\n" +
```

Comment:
If configs can pull in secrets, doesn't this have security consequences since configs are not on tmpfs?

Reply:
that seems really complicated, why don't we move all configs to tmpfs? They should be accounted for under the container's memory usage so I don't think thats an issue.

Reply:
That would be fine with me. I don't have time to update the PR right now, but it's okay with me if someone wants to carry this.